client/src/pages/LoginPage.js

```diff
@@ -31,6 +38,42 @@ export default function LoginPage() {
     setPassword('');
   }
 
+  async function loadUser() {
+    const auth = localStorage.getItem('token');
+    try {
+      const res = await fetch('http://localhost:7100/api/auth', {
+        method: 'GET',
+        headers: {
+          'Content-Type': 'application/json',
+          'X-Auth-Token': auth
+        }
+      });
+      const me = await res.json();
+      setLogin(true);
+      setUser([
+        {
+          name: me.name
+        }
+      ]);
+      console.log(me.name);
+      console.log(user);
```

Comment:
I think you expect the updated `user` here because you called `setUser` earlier. 
You will get the old user here.
`setUser` is async and `user` is updated on the next render.

You can move your `console.log(user)` outside of this function to verify, that the user is updated.